sklearn/preprocessing/data.py

```diff
@@ -2873,3 +2876,212 @@ def inverse_transform(self, X):
+
+    handle_greater : str, 'warn' or 'error' or 'clip', default='warn'
+        Whether to raise an error or clip or warn if an
+        ordinal feature >= n_values is passed in.
+
+    Attributes
+    ----------
+    feature_indices_ : array of shape (n_features,)
+        Indices to feature ranges.
+        Feature ``i`` in the original data is mapped to features
+        from ``feature_indices_[i]`` to ``feature_indices_[i+1]``
+
+    n_values_ : array of shape (n_features,)
+        Maximum number of values per feature.
+
+    Examples
+    --------
+    Given a dataset with three features and four samples, we let the encoder
+    find the maximum value per feature and transform the data to a binary
+    unary encoding.
+
+    >>> from sklearn.preprocessing import UnaryEncoder
+    >>> enc = UnaryEncoder()
+    >>> enc.fit([[0, 0, 3],
+    ...          [1, 1, 0],
+    ...          [0, 2, 1],
+    ...          [1, 0, 2]])  # doctest: +ELLIPSIS
+    UnaryEncoder(dtype=<... 'numpy.float64'>, handle_greater='warn',
+           n_values='auto', ordinal_features='all', sparse=False)
+    >>> enc.n_values_
+    array([2, 3, 4])
+    >>> enc.feature_indices_
+    array([0, 1, 3, 6])
+    >>> enc.transform([[0, 1, 2]])
+    array([[ 0.,  1.,  0.,  1.,  1.,  0.]])
+
+    See also
+    --------
+    sklearn.preprocessing.OneHotEncoder: encodes categorical integer features
+      using a one-hot aka one-of-K scheme.
+    """
+    def __init__(self, n_values="auto", ordinal_features="all",
+                 dtype=np.float64, sparse=False, handle_greater='warn'):
+        self.n_values = n_values
+        self.ordinal_features = ordinal_features
+        self.dtype = dtype
+        self.sparse = sparse
+        self.handle_greater = handle_greater
+
+    def fit(self, X, y=None):
+        """Fit UnaryEncoder to X.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_feature]
+            Input array of type int.
+            All feature values should be non-negative otherwise will raise a
+            ValueError.
+        """
+        _transform_selected(X, self._fit, self.ordinal_features, copy=True)
+        return self
+
+    def _fit(self, X):
+        """Assumes X contains only ordinal features."""
+        X = check_array(X, dtype=np.int)
+        if self.handle_greater not in ['warn', 'error', 'clip']:
+            raise ValueError("handle_greater should be either 'warn', 'error' "
+                             "or 'clip' got %s" % self.handle_greater)
+        if np.any(X < 0):
+            raise ValueError("X needs to contain only non-negative integers.")
+        n_samples, n_features = X.shape
+
+        if (isinstance(self.n_values, six.string_types) and
+                self.n_values == 'auto'):
+            n_values = np.max(X, axis=0) + 1
+        elif isinstance(self.n_values, numbers.Integral):
+            if (np.max(X, axis=0) >= self.n_values).any():
```

Comment:
Isn't it redundant with the check at this end of `_fit`?
which is better since it uses `handle_greater`